awx/main/tests/functional/api/test_workflow_node.py

```diff
@@ -56,6 +68,90 @@ def test_node_accepts_prompted_fields(inventory, project, workflow_job_template,
          user=admin_user, expect=201)
 
 
+@pytest.mark.django_db
+class TestApprovalNodes():
+    def test_approval_node_creation(self, post, approval_node, admin_user):
+        url = reverse('api:workflow_job_template_node_create_approval',
+                      kwargs={'pk': approval_node.pk, 'version': 'v2'})
+        post(url, {'name': 'Test', 'description': 'Approval Node', 'timeout': 0},
+             user=admin_user, expect=200)
+
+        approval_node = WorkflowJobTemplateNode.objects.get(pk=approval_node.pk)
+        assert isinstance(approval_node.unified_job_template, WorkflowApprovalTemplate)
+        assert approval_node.unified_job_template.name=='Test'
+        assert approval_node.unified_job_template.description=='Approval Node'
+        assert approval_node.unified_job_template.timeout==0
+
+    def test_approval_node_creation_failure(self, post, approval_node, admin_user):
+        url = reverse('api:workflow_job_template_node_create_approval',
+                      kwargs={'pk': approval_node.pk, 'version': 'v2'})
+        post(url, {'name': '', 'description': 'Approval Node', 'timeout': 0},
+             user=admin_user, expect=400)
+        # Leave off a required param to assert that you get a 400
+        approval_node = WorkflowJobTemplateNode.objects.get(pk=approval_node.pk)
+        assert isinstance(approval_node.unified_job_template, WorkflowApprovalTemplate) is False
+
+    @pytest.mark.parametrize("is_admin, is_org_admin, status", [
+        [True, False, 200],
+        [False, False, 403],
+        [False, True, 200],
+    ])
+    def test_approval_node_creation_rbac(self, post, approval_node, alice, is_admin, is_org_admin, status):
+        url = reverse('api:workflow_job_template_node_create_approval',
+                      kwargs={'pk': approval_node.pk, 'version': 'v2'})
+        if is_admin is True:
+            approval_node.workflow_job_template.admin_role.members.add(alice)
+        if is_org_admin is True:
+            approval_node.workflow_job_template.organization.admin_role.members.add(alice)
+        post(url, {'name': 'Test', 'description': 'Approval Node', 'timeout': 0},
+             user=alice, expect=status)
+
+    @pytest.mark.django_db
+    def test_approval_node_exists(self, post, approval_node, admin_user, get):
+        workflow_job_template = WorkflowJobTemplate.objects.create()
+        approval_node = WorkflowJobTemplateNode.objects.create(
+            workflow_job_template=workflow_job_template
+        )
+        url = reverse('api:workflow_job_template_node_create_approval',
+                      kwargs={'pk': approval_node.pk, 'version': 'v2'})
+        post(url, {'name': 'URL Test', 'description': 'An approval', 'timeout': 0},
```

Comment:
You might not need this test, since it arguably is testing the same thing as https://github.com/ansible/awx/pull/4264/commits/2f1ae2a7e30d256bc8be3ad6f16af4d0d2c1bc39#diff-88e3eee88f4c90d64a4617eff0495b7eR73